src/libs/data-table-provenance.js

```diff
@@ -0,0 +1,90 @@
+import _ from 'lodash/fp'
+import { useCallback, useState } from 'react'
+import { Ajax } from 'src/libs/ajax'
+import { reportError } from 'src/libs/error'
+import { useCancellation } from 'src/libs/react-utils'
+import * as Utils from 'src/libs/utils'
+
+
+export const maxSubmissionsQueriedForProvenance = 25
+
+const getSubmissionsWithRootEntityType = async (workspace, entityType, { signal } = {}) => {
+  const { workspace: { namespace, name } } = workspace
+
+  // The root entity type is stored on method configurations, not submissions. To avoid fetching
+  // every submission's configuration, first use the list submissions endpoint and filter by
+  // submission entity type.
+  //
+  // If a workflow is run on a single entity, the submission entity type will be the root entity type.
+  // If a workflow is run on multiple entities, the submission entity type will be a set of the root entity type.
+  //
+  // After fetching method configurations, another filter is required remove submissions whose root entity type
+  // was a set of the desired entity type.
+  const allSubmissions = await Ajax(signal).Workspaces.workspace(namespace, name).listSubmissions()
+  const submissionsMaybeUsingEntityType = _.flow(
+    _.filter(submission => {
+      const submissionEntityType = submission.submissionEntity.entityType
+      return submissionEntityType === entityType || submissionEntityType === `${entityType}_set`
+    }),
+    _.slice(0, maxSubmissionsQueriedForProvenance)
+  )(allSubmissions)
+
+  const submissionConfigurations = await Promise.all(
+    _.map(
+      submission => Ajax(signal).Workspaces.workspace(namespace, name).submission(submission.submissionId).getConfiguration(),
+      submissionsMaybeUsingEntityType
+    )
+  )
+
+  return _.flow(
+    _.map(([submission, configuration]) => ({ submission, configuration })),
+    _.filter(({ configuration }) => configuration.rootEntityType === entityType),
+    _.orderBy(({ submission }) => new Date(submission.submissionDate), 'desc')
+  )(_.zip(submissionsMaybeUsingEntityType, submissionConfigurations))
+}
+
+const getColumnFromOutputExpression = expr => {
+  const match = /^this\.([^.]+)$/.exec(expr)
+  return match ? match[1] : null
+}
+
+const getColumnProvenance = async (workspace, entityType, { signal } = {}) => {
+  const submissions = await getSubmissionsWithRootEntityType(workspace, entityType, { signal })
+
+  return _.flow(
+    _.flatMap(({ submission, configuration }) => _.map(([output, expression]) => ({ submission, configuration, output, expression }), _.toPairs(configuration.outputs))),
+    _.map(({ submission, configuration, output, expression }) => ({ submission, configuration, output, column: getColumnFromOutputExpression(expression) })),
+    _.filter(({ column }) => !!column),
```

Comment:
Consider whether this might be a clearer way of saying, "remove any items where this key has no value:"
```suggestion
    _.filter(x => !!x.column),
```

Reply:
Changed in https://github.com/DataBiosphere/terra-ui/commit/959268e1cfa7295c1d3848b5e4f86a37413fdb13.